src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -143,9 +143,12 @@ public async Task InitializeAsync(ChainedHeader chainTip)
 
             uint256 consensusTipHash = await this.consensusRules.GetBlockHashAsync().ConfigureAwait(false);
 
+            // If we are on a network with a maximum reorg setting, don't bother searching further back than the limit.
```

Comment:
I'm not sure this is correct. The initialisation should be allowed to go beyond finalised the block. Also note that all the code should work with finalised block instead of maximum reorganisation directly. But such limitation should not be here

Reply:
@zeptin I prefer if in this PR we only add the code and not activate it, less problems to worry about if something is broken so please revert change to CM

Reply:
Yes, I think that is better. I will do that shortly.

Reply:
Done